### PR TITLE
remove job from pull and import

### PR DESCRIPTION
**Pull request by dalanlan**
part of issue #12151 
remove job from pull and import 
Signed-off-by: Simei He hesimei@zju.edu.cn

### CODE REVIEW

**Comment by dalanlan** [comment]
@tiborvass

**Comment by cpuguy83** [comment]
Tests are failing

File: graph/pull.go [hunk 1]
```diff
@@ -65,7 +58,7 @@ func (s *TagStore) CmdPull(job *engine.Job) error {
 		return err
 	}
 
-	r, err := registry.NewSession(authConfig, registry.HTTPRequestFactory(metaHeaders), endpoint, true)
+	r, err := registry.NewSession(imagePullConfig.AuthConfig, registry.HTTPRequestFactory(imagePullConfig.MetaHeaders), endpoint, true)
```
Comment (calavera) [inline graph/pull.go]: I think the session assumes that AuthConfig is never nil, you'll need to initialize it with `&registry.AuthConfig{}`

**Comment by LK4D4** [comment]
Carried in #12490

**Pull request closed without merging**